delegate/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+use anchor_lang::{prelude::*, InstructionData};
+use anchor_spl::token::{transfer, Transfer, Token, TokenAccount, Approve, approve};
+use solana_program::{instruction::Instruction, program_option::COption};
+
+// TODO update with correct ID
+declare_id!("2Ls5MquEmp42AXBxKXX3a9Gu54aPYYVC19tV7RCMKsTt");
+
+#[account]
+pub struct SwapInfo {
+    pub is_initialized: bool,
+    pub poster: Pubkey,
+    pub poster_sell_account: Pubkey,
+    pub poster_buy_account: Pubkey,
+    pub poster_sell_amount: u64,
+    pub poster_buy_amount: u64,
+}
+
+pub const SWAP_INFO_BYTES: usize = 1 + 32 + 32 + 32 + 8 + 8;
+
+#[derive(Accounts)]
+pub struct PostSwap<'info> {
+    #[account(mut)]
+    pub poster: Signer<'info>,
+    #[account(
+        mut, 
+        constraint = sell_from.owner == poster.key(),
+    )]
+    pub sell_from: Account<'info, TokenAccount>,
+    pub buy_to: Account<'info, TokenAccount>,
+    #[account(
+        init_if_needed,
+        space = 8 + SWAP_INFO_BYTES,
+        payer=poster,
+        owner=*program_id,
+        seeds=[sell_from.key().as_ref()],
+        bump
+    )]
+    pub swap_info: Account<'info, SwapInfo>,
+    pub delegate_program: Program<'info, program::Delegate>,
+    pub token_program: Program<'info, Token>,
+    pub system_program: Program<'info, System>,
+}
+
+pub fn initialize_swap(
```

Comment:
You don't need to write this code it's generated for you

Reply:
I don't think that's true. `cargo expand` doesn't show any functions like that. My understand is that functions of that nature are only created in javascript when you use `anchor build` etc. Theres no Rust equivalent generated for you. Unless I'm missing something?

Reply:
I'll send you some examples it may need a feature to be enable I don't recall exactly but let me look at some of my programs and I'll send over the examples